### PR TITLE
Make nunique in DataFrame to return a Koalas DataFrame instead of pandas'

### PR DESCRIPTION
**Pull request by HyukjinKwon**
The approach is basically similar with `DataFrame._reduce_for_stat_function`.

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -3527,7 +3527,7 @@ def nunique(
 
         Returns
         -------
-        The number of unique values per column as a pandas Series.
+        The number of unique values per column as a koalas Series.
```
Comment (itholic) [inline databricks/koalas/frame.py]: nit: `koalas` -> `Koalas`

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -3550,22 +3550,31 @@ def nunique(
...
-        return res.T.iloc[:, 0]
+        )
+
+        # The data is expected to be small so it's fine to transpose/use default index.
+        with ks.option_context(
+            "compute.default_index_type", "distributed", "compute.max_rows", None
+        ):
+            kdf = DataFrame(sdf)  # type: ks.DataFrame
+            internal = _InternalFrame(
+                kdf._internal.spark_frame,
```
Comment (itholic) [inline databricks/koalas/frame.py]: Not sure, but maybe `spark_frame=kdf._internal.spark_frame` is better?

**Comment by itholic** [comment]
LGTM except 2 nit comments 👍

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**